### PR TITLE
Ruby: Two fixes for `private` methods

### PR DESCRIPTION
**Pull request by hvitved**
1. `ModuleBase::get(A)Method` did not work for methods directly annotated with `private`.
2. Call graph construction did not take into account that `private` methods can be overridden.

Commit-by-commit review is suggested.

I sampled a couple of new call graph edges for each project, and they all looked legit.

### CODE REVIEW

**Review by aibaars — COMMENTED** [review]
Looks good to me. One question about "nested" methods,

File: ruby/ql/lib/codeql/ruby/ast/Module.qll [hunk 1]
```diff
@@ -49,24 +50,54 @@ class Module extends TModule {
...
+ * in
+ *
+ * ```rb
+ * module M
+ *   def m
+ *     def nested; end
+ *   end
+ * end
+ * ```
+ *
```
[the remark this answers is not in the document]
Comment (aibaars) [reply ruby/ql/lib/codeql/ruby/ast/Module.qll:74]: True, but you are including conditional methods like:
```
if rails.version < 6
   def x; end
end
```

[the remark this answers is not in the document]
Comment (aibaars) [reply ruby/ql/lib/codeql/ruby/ast/Module.qll:74]: I'm fine with excluding nested method for now. Just wanted to make sure it was for the right reason.

[the remark this answers is not in the document]
Comment (hvitved) [reply ruby/ql/lib/codeql/ruby/ast/Module.qll:74]: > True, but you are including conditional methods like

Indeed. Let's add all nested methods in a follow-up PR.

**Review by aibaars — APPROVED** [review]

**Pull request merged**